runtime/lua/vim/treesitter.lua

```diff
@@ -61,7 +61,7 @@ function M._create_parser(bufnr, lang, opts)
     { on_bytes = bytes_cb, on_detach = detach_cb, on_reload = reload_cb, preview = true }
   )
 
-  self:parse()
+  self:async_parse()
```

Comment:
This is a bit dangerous because we return `self` right after this- imo this function should not parse at all but I would like to hear some feedback on this

(I tested removing `parse()` completely and not many tests failed, but it would mean that uses would have to manually parse in more cases than usual, e.g. when using `get_node()`)

Reply:
> This is a bit dangerous because we return `self` right after this

What is the worst case scenario?

If we want to make it safe, we could proxy the self methods to force waiting on the parse result. But I guess that could force synchronous behavior in more places than we want? Or throw an error if a self method is called before parse is finished.

Reply:
I think (at least for now) we need to distinguish between "background" processing -- such as highlighting -- and "on demand" processing -- such as textobjects. Async is absolutely necessary for the former but can be deferred for the latter.

So I'd focus on highlighting performance and everything else not breaking; we can then make those properly async one by one (where it makes sense).

Reply:
I agree. In this case the bare minimum to allow async highlighting is this change (in `_create_parser`, either running an async parse or removing that parse step altogether) and then the other one I have in `highlighter.new`

Reply:
This is a good thread to leave un-resolved, for future reference.